test/extensions/common/aws/sigv4a_signer_impl_test.cc

```diff
@@ -0,0 +1,442 @@
+                        credentials_.secretAccessKey()->size()));
+  SigV4AKeyDerivation::derivePublicKey(ec_key);
+  Envoy::Logger::Registry::setLogLevel(spdlog::level::debug);
+  Http::TestRequestHeaderMapImpl headers{};
+
+  headers.setMethod("GET");
+  headers.setPath("/");
+  headers.addCopy(Http::LowerCaseString("host"), "www.example.com");
+
+  // Sign the message using our signing algorithm
+  signer_.signUnsignedPayload(headers, "ap-southeast-2");
+
+  // Now manually sign the same string_to_sign
+  std::string canonical_request = R"EOF(GET
+/
+
+host:www.example.com
+x-amz-content-sha256:UNSIGNED-PAYLOAD
+x-amz-date:20180102T030400Z
+x-amz-region-set:ap-southeast-2
+
+host;x-amz-content-sha256;x-amz-date;x-amz-region-set
+UNSIGNED-PAYLOAD)EOF";
+
+  std::string short_date = "20180102";
+  std::string credential_scope = fmt::format(fmt::runtime("{}/service/aws4_request"), short_date);
+  std::string long_date = "20180102T030400Z";
+  std::string string_to_sign =
+      fmt::format(fmt::runtime(SigV4ASignatureConstants::get().SigV4AStringToSignFormat), long_date,
+                  credential_scope,
+                  Hex::encode(crypto_util.getSha256Digest(Buffer::OwnedImpl(canonical_request))));
+  auto hash = crypto_util.getSha256Digest(Buffer::OwnedImpl(string_to_sign));
+  // Extract the signature that is generated
+  EXPECT_THAT(
+      headers.get(Http::CustomHeaders::get().Authorization)[0]->value().getStringView(),
+      testing::StartsWith("AWS4-ECDSA-P256-SHA256 Credential=akid/20180102/service/aws4_request, "
+                          "SignedHeaders=host;x-amz-content-sha256;x-amz-date;x-amz-region-set, "
+                          "Signature="));
+  std::vector<std::string> v = absl::StrSplit(
+      headers.get(Http::CustomHeaders::get().Authorization)[0]->value().getStringView(),
+      "Signature=");
+  std::vector<uint8_t> signature = Hex::decode(v[1]);
+  // Check that the signature generated by our algorithm can be verified by the matching public key
+  EXPECT_EQ(1,
+            ECDSA_verify(0, hash.data(), hash.size(), signature.data(), signature.size(), ec_key));
+}
+
+TEST_F(SigV4ASignerImplTest, SignAndVerifyUnsignedPayloadMultiRegion) {
+  EXPECT_CALL(*credentials_provider_, getCredentials()).WillOnce(Return(credentials_));
+  auto& crypto_util = Envoy::Common::Crypto::UtilitySingleton::get();
+
+  EC_KEY* ec_key = SigV4AKeyDerivation::derivePrivateKey(
+      absl::string_view(credentials_.accessKeyId()->data(), credentials_.accessKeyId()->size()),
+      absl::string_view(credentials_.secretAccessKey()->data(),
+                        credentials_.secretAccessKey()->size()));
+  SigV4AKeyDerivation::derivePublicKey(ec_key);
+
+  Http::TestRequestHeaderMapImpl headers{};
+
+  headers.setMethod("GET");
+  headers.setPath("/");
+  headers.addCopy(Http::LowerCaseString("host"), "www.example.com");
+
+  // Sign the message using our signing algorithm
+  signer_.signUnsignedPayload(headers, "ap-southeast-2,*");
```

Comment:
what is different between `ap-southeast-2,*` and just `*`, wouldn't wildcard `*` cover `ap-southeast-2` as well?